Add image data structure and matrix multiplication algorithms

IzmaylovI opened the pull request:
1) Added a description of the project in the Readme.md
2) A data structure has been created to store the image in the form (bgr/rgb). Implemented parsing of BMP formats without third-party dependency
3) Matrix multiplication functions have been added using intrinsics NEON and OpenMP

IzmaylovI committed:
Update README.md

IzmaylovI committed:
Update README.md

IzmaylovI committed:
Update README.md

IzmaylovI committed:
Create README.md

IzmaylovI committed:
Update README.md

IzmaylovI committed:
Update README.md

IzmaylovI committed:
first_request

IzmaylovI committed:
Update README.md

aobolensk commented on CMakeLists.txt:
```diff
@@ -0,0 +1,21 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.25.0)
+
+set(CMAKE_CXX_COMPILER "C:/MinGW/bin/g++.exe")
+set(CMAKE_C_COMPILER "C:/MinGW/bin/gcc.exe")
```
This solution is not universal, at least do not override C/CXX compiler if it is already found

aobolensk commented on headers/Color.hpp:
```diff
@@ -0,0 +1,163 @@
+#ifndef _COLOR_
+#define _COLOR_
+#include <iostream>
+enum mode { rgb, bgr };
```
It is not the mode in fact, it could be called "layout"

aobolensk commented on headers/Function.hpp:
```diff
@@ -0,0 +1,199 @@
+    return res;
+}
+
+/*
```
Do we need this comment?

aobolensk commented on headers/Structer.hpp:
```diff
@@ -0,0 +1,239 @@
+    int height;
+    int width;
+
+    Mtrx(int height_m = 0, int width_m = 0, ValType element = 0);
```
Why do we have default width/height set to 0?

IzmaylovI committed:
First_correction

IzmaylovI committed:
Merge branch 'main' of https://github.com/IzmaylovI/itlab_2022

IzmaylovI commented:
Fixed the code:
1) removed the override C/CXX compiler.
2) renamed mode to layout
3) removed the commentaries in the file Function.hpp. For matrices of elements of the base type, everything will work. For pixel matrices, one more function needs to be implemented
4) removed the default value of 0 in width/height because there is no need for this

allnes commented on CMakeLists.txt:
```diff
@@ -0,0 +1,18 @@
+    message(STATUS "Not link OpenMP")
+endif()
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
```
please set OpenMP's flags when `OpenMP_FOUND` is true

allnes commented on README.md:
```diff
@@ -1 +1,19 @@
+- Working with Raspberry PI devices in practice.
+
+## Software modules:
+### Module 1. 
```
Please name the modules according to their purpose

allnes commented on headers/Structer.hpp:
```diff
@@ -0,0 +1,239 @@
+#include "Color.hpp"
+
+template <typename ValType>
+class Mtrx {
```
I think this class should be made purely abstract. 
@aobolensk how do you think?

IzmaylovI committed:
BMP_parsing

allnes commented:
@IzmaylovI please correct commit changes, old changes is lost

IzmaylovI committed:
Corrected_commit

IzmaylovI commented:
Corrected the commit. Is there anything else that needs to be corrected?

aobolensk commented:
Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM

aobolensk commented:
@allnes please, use squash strategy when we will be merging this PR

IzmaylovI commented:
> Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM


where do we need to change the description?

aobolensk commented:
> > Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM
> 
> 
> where do we need to change the description?

Description of this pull request

IzmaylovI committed:
fixed_Function.hpp

IzmaylovI commented:
1) Removed extra lines in CMakeLists.txt
2) Wrote non-template matrix multiplication functions using intrinsics
3) Тame the modules according to their purpose in Readme.md
4) Set OpenMP's flags when "OpenMP_FOUND" is true  in CMakeLists.txt
5) Deleted unnecessary file - BMP.cpp 
6) Removed unnecessary comments in files
